### PR TITLE
Add support for `aarch64`

### PR DESCRIPTION
**Pull request by xRuiAlves**
Currently the setup action only supports the `x86_64` architecture.

`coursier` also publishes binaries for `aarch64` (_e.g._, [v2.1.0-RC4](https://github.com/coursier/coursier/releases/tag/v2.1.0-RC4)).

Leveraging the system's `process.arch` to choose which file architecture to download.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -23,28 +29,28 @@ async function execOutput(cmd: string, ...args: string[]): Promise<string> {
...
-  const baseUrl = `https://github.com/coursier/coursier/releases/download/v${csVersion}/cs-x86_64`
+  const baseUrl = `https://github.com/coursier/coursier/releases/download/v${csVersion}/cs-${architecture}`
   let csBinary = ''
   switch (process.platform) {
     case 'linux': {
       const guid = await tc.downloadTool(`${baseUrl}-pc-linux.gz`)
-      const arc = `${guid}.gz`
-      await cli.exec('mv', [guid, arc])
-      csBinary = arc
+      const archive = `${guid}.gz`
```
Comment (xRuiAlves) [inline src/main.ts:37]: Renamed these `arc` variables to `archive` - With the introduction of the new `architecture` variable, `arc` could be ambiguous, and the renaming enhances clarity on this.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -12,6 +12,7 @@ jobs:
       matrix:
         os: [ubuntu-latest, macOS-latest, windows-latest]
         jvm: ['8', '11']
+        architecture: ['x86_64', 'aarch6']
```
Comment (alexarchambault) [inline .github/workflows/test.yml:15]: Do you really mean `aarch6` rather than `aarch64`? (`aarch6` also appears a few times below)

**Review by alexarchambault — COMMENTED** [review]
Thanks!

There's no need to update things under `dist`, the CI will handle it on its own (it updates things locally, and later opens PRs to update things in `main` if needed). The changes there can be removed.

I also left one comment (might be more of a question)

File: .github/workflows/test.yml (hunk 2 above)
Comment (xRuiAlves) [reply .github/workflows/test.yml]: Thanks @alexarchambault, nice catch! missed that last `4` in the process of copy-pasting. I've also reverted the changes in `dist`.

**Comment by xRuiAlves** [comment]
Additionally, I've refactored the action logic to make use of the system's `process.arch` instead of relying on the user specifying the architecture. I will update the PR description accordingly.

**Comment by xRuiAlves** [comment]
Hey @alexarchambault! Was wondering if you had a chance to taking a look at this?

**Comment by xRuiAlves** [comment]
Hey @alexarchambault! Was wondering if you had a chance to taking a look at this in the past few weeks?

**Review by alexarchambault — APPROVED** [review]
Thanks @xRuiAlves!

**Pull request merged**